enhancements/microshift/microshift-sriov-operator.md

```diff
@@ -0,0 +1,246 @@
+---
+title: microshift-sriov-operator
+authors:
+  - vanhalenar
+reviewers:
+  - "@pacevedom"
+  - "@pmtk"
+  - "@ggiguash"
+approvers:
+  - "@jerpeter1"
+api-approvers:
+  - None
+creation-date: 2025-09-19
+last-updated: 2025-09-19
+tracking-link:
+  - https://issues.redhat.com/browse/OCPSTRAT-2329
+---
+
+# MicroShift SR-IOV operator integration
+
+## Summary
+
+The following enhancement proposes integrating the OpenShift SR-IOV network
+operator into MicroShift in order to enable users to configure VFs in a more
+convenient manner.
+
+## Motivation
+
+Currently, SR-IOV can be used with MicroShift by manually configuring VFs
+(Virtual Functions) on a PF (Physical Function) on OS level and using multus to
+map the VFs to pods. This approach is tedious and non-idiomatic. By using the
+SR-IOV network operator, users can configure VFs in a declarative way by
+specifying and deploying a NetworkNodePolicy CRD.
+
+### User Stories
+
+As a MicroShift admin, I want to configure VFs on PFs in a declarative manner.
+
+As a MicroShift admin, I don't want to configure VFs manually on OS level, I
+want to use higher level tools.
+
+### Goals
+
+- Enable users to create VFs declaratively, instead of manually configuring
+  them, by providing the OpenShift SR-IOV network operator as an optional RPM. 
+- Provide a simple smoke test to check that the pods were deployed correctly,
+  and that the VFs can be assigned.
+
+### Non-Goals
+
+- Provide a full test of all SR-IOV functions. The assumption is that OpenShift
+  testing of SR-IOV covers the MicroShift use case as well.
+
+## Proposal
+
+- Provide SR-IOV network operator for MicroShift as an optional RPM containing
+  required manifests that will be applied upon MicroShift starting. 
+- Manifests will be based on existing manifests for OpenShift SR-IOV operator.
+  Possible changes may include optimizations in regards to CPU/memory usage (to
+  be discussed).
```

Comment:
Can we link the discussion on this topic from below? Currently, it's not clear if the discussion is part of the enhancement or not.